src/connector/src/parser/scalar_adapter.rs

```diff
@@ -185,56 +185,52 @@ impl ScalarAdapter<'_> {
         }
     }
 
-    // return scalar and a flag indicating if the list should fallback to NULL
-    pub fn into_scalar_in_list(self, ty: DataType) -> (Option<ScalarImpl>, bool) {
-        match (&self, &ty) {
-            (ScalarAdapter::Builtin(scalar), _) => (Some(scalar.into_scalar_impl()), false),
-            (ScalarAdapter::Uuid(uuid), &DataType::Varchar) => {
-                (Some(ScalarImpl::from(uuid.to_string())), false)
-            }
-            (ScalarAdapter::Numeric(numeric), &DataType::Varchar) => {
-                if pg_numeric_is_special(numeric) {
-                    (None, true)
-                } else {
-                    (Some(ScalarImpl::from(pg_numeric_to_string(numeric))), false)
+    pub fn build_scalar_in_list(
```

Comment:
Maybe we can refactor this func to unify the interfere to `into_scalar` in future, sine List is also a scalar in RW.

Reply:
Let's do it in next PR

Reply:
https://github.com/risingwavelabs/risingwave/pull/16714/files PTAL